openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/resources/TemplateProcessor.java

```diff
@@ -0,0 +1,40 @@
+package org.arquillian.cube.openshift.impl.resources;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.arquillian.cube.openshift.api.Template;
+import org.arquillian.cube.openshift.api.model.DeploymentConfig;
+import org.arquillian.cube.openshift.api.model.OpenShiftResource;
+import org.arquillian.cube.openshift.impl.adapter.OpenShiftAdapter;
+import org.arquillian.cube.openshift.impl.utils.Operator;
+import org.arquillian.cube.openshift.impl.utils.ParamValue;
+import org.arquillian.cube.openshift.impl.utils.TemplateUtils;
+
+public interface TemplateProcessor {
+
+    List<List<? extends OpenShiftResource>> processTemplateResources();
```

Comment:
Can we flat this to be a single List of `OpenShiftResources` instead of list of lists?

Reply:
We have list of generics due to which it's not possible. See ref [[1]](https://docs.oracle.com/javase/tutorial/extra/generics/wildcards.html), [[2]](https://stackoverflow.com/questions/6504208/another-java-generic-question/6504576#6504576)

Reply:
I was wrong here. We can do it by contravariance( widening a reference)